packages/Range/_.candy

```diff
@@ -0,0 +1,103 @@
+bound = use ".bound"
+[bool, ifElse, int, iterator] = use "Core"
+[print] = use "Builtins"
+
+is range := range %
+  Empty -> True
+  All -> True
+  [start, end] -> bound.is start | bool.and (bound.is end)
+  _ -> False
+
+from a :=
+  needs (int.is a)
+  [Start: a, End: Unbounded]
+to a b :=
+  needs (int.is a)
+  needs (int.is b)
+  [Start: Inclusive a, End: Exclusive b]
+until a b :=
+  needs (int.is a)
+  needs (int.is b)
+  [Start: Inclusive a, End: Inclusive b]
+
+normalize range :=
```

Comment:
Alternatively, we could enforce ranges to always be normalized. What do you think?